rust/core-lib/src/editor.rs

```diff
@@ -302,9 +304,14 @@ impl<W: Write + Send + 'static> Editor<W> {
         }
     }
 
+    fn has_unsaved_changes(&self) -> bool {
```

Comment:
I wonder if it wouldn't make more sense to store this as a member of the editor struct, or somewhere like in `tab_ctx` (which gets automatically passed to the view when we render). Then we could compute it only after an edit occurs, instead of on each render cycle.

I like the idea of this living in something _like_ `tab_ctx`, because it saves it having to be passed to each call? I wonder if this isn't also true of the `style_spans`? But I'm not sure if that makes sense.